model/sharedStimulus/service/UpdateService.php

```diff
@@ -0,0 +1,148 @@
+
+        $fileName = $this->getTempFileName();
+        $filePath = $this->saveTemporaryFile($fileName, $newBody);
+
+        $this->validateResource($resource);
+        $this->validateXml($filePath);
+
+        $this->getMediaService()->editMediaInstance($filePath, $id, null, $userId);
+
+        return new SharedStimulus(
+            $id,
+            $resource->getLabel(),
+            $resource->getOnePropertyValue($this->getProperty(MediaService::PROPERTY_LANGUAGE))->getUri(),
+            $newBody
+        );
+    }
+
+
+    private function getTempFileName(): string
+    {
+        return 'shared_stimulus_' . uniqid() . '.xml';
+    }
+
+    private function getMediaService(): MediaService
+    {
+        return MediaService::singleton();
+    }
+
+    /**
+     * @throws ErrorException
+     */
+    private function saveTemporaryFile(string $fileName, string $templateContent): string
+    {
+        $fileDirectory = $this->getOption(self::OPTION_TEMP_UPLOAD_PATH) ?? sys_get_temp_dir();
+
+        $filePath = $fileDirectory . DIRECTORY_SEPARATOR . $fileName;
+
+        if (!is_writable($fileDirectory) || file_put_contents($filePath, $templateContent) === false) {
+            throw new ErrorException(
+                sprintf(
+                    'Could not save Shared Stimulus to temporary path %s',
+                    $filePath
+                )
+            );
+        }
+
+        return $filePath;
+    }
+
+    /**
+     * @param Resource $resource
+     */
+    private function validateResource(Resource $resource): void
+    {
+        if (!$resource->isInstanceOf($this->getClass(MediaService::ROOT_CLASS_URI))) {
+            $this->logAlert(
+                sprintf(
+                    'Incorrect resource provided, %s should be subtype of  %s',
+                    $resource->getUri(),
+                    MediaService::ROOT_CLASS_URI
+                )
+            );
+            throw new InvalidArgumentException('Invalid resource provided');
+        }
+    }
+
+    private function validateXml(string $fileName): void
+    {
+        try {
+            SharedStimulusImporter::isValidSharedStimulus($fileName);
+        } catch (XmlStorageException $e) {
+            $this->logAlert(sprintf('Incorrect shared stimulus xml, %s', $e->getMessage()));
+            throw new InvalidArgumentException('Invalid XML provided');
+        }
+
+        $resolver = new TaoMediaResolver();
+        $xmlDocument = new XmlDocument();
+        $xmlDocument->load($fileName, false);
+
+        $images = $xmlDocument->getDocumentComponent()->getComponentsByClassName('img');
+
+        foreach ($images as $image) {
+            $source = $image->getSrc();
+            if (false === strpos($source, 'data:image')) {
+                $asset = $resolver->resolve($source);
+                if ($asset instanceof MediaAsset) {
+                    $info = $asset->getMediaSource()->getFileInfo($asset->getMediaIdentifier());
+                }
```

Comment:
```suggestion
                $info = $asset->getMediaSource()->getFileInfo($asset->getMediaIdentifier());
```
This check is unnecessary the method always returns an `MediaAsset` or throw an exception. But if what @siwane said is true and we are including the QTI item as dependency we can validate it using existent mechanism.

Reply:
That was a bit of typo with `MediaAsset` ( fixed ), but it have not same logic as we do in packer, as here I'm checking if taomedia managed specificly assets exists in the system.